test/polymer/behavior-finder_test.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+"use strict";
+
+const assert = require('chai').assert;
+const fs = require('fs');
+const path = require('path');
+
+const BehaviorDescriptor = require('../../lib/ast/behavior-descriptor').BehaviorDescriptor;
+const BehaviorFinder = require('../../lib/polymer/behavior-finder').BehaviorFinder;
+const JavaScriptParser = require('../../lib/javascript/javascript-parser').JavaScriptParser;
+
+suite('BehaviorFinder', () => {
+
+  let document;
+  let analyzer;
+  let behaviors;
+  let behaviorsList;
+
+  suiteSetup(() => {
+    let parser = new JavaScriptParser();
+    let file = fs.readFileSync(
+        path.resolve(__dirname, '../static/js-behaviors.js'), 'utf8');
+    document = parser.parse(file, '/static/js-behaviors.js');
+    let finder = new BehaviorFinder();
+    let visit = (visitor) => document.visit([visitor]);
+
+    return finder.findEntities(document, visit)
+      .then((entities) => {
+        behaviors = new Map();
+        behaviorsList = entities.filter((e) => e instanceof BehaviorDescriptor);
+        for (let behavior of behaviorsList) {
+          behaviors.set(behavior.is, behavior);
```

Comment:
I'm not sure that's so needed in the test. We know the input and check for the existence of all the behaviors defined there.